File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    const idMap = new Map(id_list.map(id => [id, {
        reported: [],
        emails: 0
    }]));
    
    report.forEach(r => {
        const [re, ed] = r.split(" ");
        
        const obj = idMap.get(ed);
        
        const set = new Set(obj.reported);
        
        set.add(re);
        
        idMap.set(ed, {...obj, reported: [...set]});
    })
    
    for(const id of id_list){
        const reports = idMap.get(id)['reported'];
        
        if(reports.length >= k){
            for(const re of reports){
                const obj = idMap.get(re);
                idMap.set(re, {...obj, emails: obj.emails + 1})
            }
        } 
    }
    
    return id_list.map(id => idMap.get(id)['emails']);
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    report_dict = {id: set() for id in id_list}
    reported_dict = {id: 0 for id in id_list}
    
    for r in report:
        [report_id, reported_id] = r.split()
        
        # 신고 여부
        has_reported = reported_id in report_dict[report_id]
        if has_reported == False:
            report_dict[report_id].add(reported_id)
            reported_dict[reported_id] += 1
    
    
    result = []
    for id in id_list:
        count = 0
        for reported_id in report_dict[id]:
            if reported_dict[reported_id] >= k:
                count += 1
        result.append(count)
        
    return result